lib/commands/exam.js

```diff
@@ -254,5 +370,87 @@ module.exports = TestCommand.extend({
       case 'json':
         return fs.readJsonSync(file);
     }
+  },
+
+  /**
+   * Adds additional event handlers to config to enable load balancing.
+   * 
+   * @param {object} config
+   * @return {object} events
+   */
+  _addLoadBalancingBrowserSocketEvents: function(config) {
+    const events = config.additional_browser_socket_events || {};
+
+    events['set-modules-queue'] = function (modules) {
+      // When `load-balance` option is valid we want to have one static list of modules on server side to send a module path to browsers.
+      const proto = Object.getPrototypeOf(this);
+
+      // browserModuleMapping is defined if `replay-execution` option is being used.
+      const browserModuleMapping = this.config.progOptions.browser_module_mapping;
+
+      // Updating browserModuleMapping which maps from browser ids to a list of AMD modules ran per the browser
+      // is needed when running test suite in ci.
+      if (this.config.appMode === 'ci' && browserModuleMapping && !this.module) {
+        const browserId = /_browser=\s*(.*)/.exec(this.launcher.settings.test_page)[1];
+        this.moduleQueue = browserModuleMapping[browserId];
+      } else if (!proto.moduleQueue) {
+        proto.moduleQueue = modules;
+        proto.moduleMap = {};
+      }
+    }
+
+    events['get-next-module'] = function () {
+      const proto = Object.getPrototypeOf(this);
+      const replayExecution = this.config.progOptions.browser_module_mapping;
+      const moduleQueue = this.moduleQueue;
+      const moduleName = moduleQueue.shift();
+
+      if (!moduleName) {
+        this.socket.emit('module-queue-complete');
+      } else {
+        this.socket.emit('next-module-response', moduleName);
+      }
+
+      // keep track of the modules ran per browserId when appMode is ci and replayExecution flag is false
+      if (this.config.appMode === 'ci' && !replayExecution) {
```

Comment:
I feel this if should be inside the `else` condition in line 410, because if `moduleName` is null or undefined, we shouldn't need to write it to the moduleMap